### PR TITLE
Improve `make_waveform`

### PR DESCRIPTION
**Pull request by dawoodkhan82**
## Description
Animates the waveforms, because I needed something to work on while the frontend files were being refactored. Tested on 35sec audio clips, and animating adds ~4 seconds (can be improved) on processing time compared to `main`.

https://github.com/gradio-app/gradio/assets/12725292/7480180e-de79-4ff4-b53c-c1a5e052b9d1

Please include a concise summary, in clear English, of the changes in this pull request. If it closes an issue, please mention it here.

Closes: #(issue)

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests & Changelog

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

1. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`
  
1. Unless the pull request is labeled with the "no-changelog-update" label by a maintainer of the repo, all pull requests must update the changelog located in `CHANGELOG.md`:

Please add a brief summary of the change to the Upcoming Release section of the `CHANGELOG.md` file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

### CODE REVIEW

File: gradio/helpers.py [hunk 1]
```diff
@@ -797,71 +798,97 @@ def get_color_gradient(c1, c2, n):
...
             np.arange(0, bar_count),
             samples * 2,
             bottom=(-1 * samples),
             width=bar_width,
             color=color,
         )
-        plt.axis("off")
-        plt.margins(x=0)
+
+        def animate(_):
```
Comment (freddyaboulton) [inline gradio/helpers.py:815]: If this increases the runtime about a 10% increase in runtime (going off of your numbers), I think it should be configurable by users! Some may not want to pay that penalty. WDYT?

**Comment by abidlabs** [comment]
Tried this out with `demo/make_waveform/run.py` and the resulting videos are definitely funner, nice! However, processing time did increase from 5s to 10s so I'd make this only run if the user sets an optional parameter.

**Comment by Muhammad2022D** [comment]
Awesome

**Comment by dawoodkhan82** [comment]
@abidlabs @freddyaboulton Yeah makes sense, I'll add a flag and make it opt-in. I also think there might be some small changes I can make to reduce the processing time.

File: gradio/helpers.py [hunk 2]
```diff
@@ -767,6 +769,7 @@ def make_waveform(
         bars_color: Color of waveform bars. Can be a single color or a tuple of (start_color, end_color) of gradient
         bar_count: Number of bars in waveform
         bar_width: Width of bars in waveform. 1 represents full width, 0.5 represents half width, etc.
+        animate_waveform: Whether to animate the waveform
```
Comment (abidlabs) [inline gradio/helpers.py:772]: Can we be more specific on what animate means here?
```suggestion
        animate_waveform: Whether to animate the waveform
```

**Review by abidlabs — APPROVED** [review]

**Comment by abidlabs** [comment]
LGTM @dawoodkhan82, just a couple of nits above

**Comment by abidlabs** [comment]
Briefly synced with @dawoodkhan82 on this. This looks ready to go, but we'll rename `animate_waveform` to `animate`. I'll make the change so we can get it out in the release

**Pull request merged**